File: MiniMaxSum/MiniMaxSum.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'miniMaxSum' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */
//Time complexity: O(nlogn) , Space complexity: O(1);
    public static void miniMaxSum(List<Integer> arr) {
    // Write your code here
        long minSum = 0L, maxSum = 0L; //use long cuz of constraints 64-bit integer
        Collections.sort(arr);
        for(int i=0; i<arr.size()-1; i++){
            minSum+=arr.get(i);
        }
        for(int i=arr.size()-1; i>=1; i--){
            maxSum+=arr.get(i);
        }
        System.out.printf("%d %d", minSum, maxSum);
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        Result.miniMaxSum(arr);

        bufferedReader.close();
    }
}

File: MiniMaxSum/MiniMaxSum.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'miniMaxSum' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function miniMaxSum(arr) {
    // Write your code here
    let totalSum = 0;
    let min = Number.MAX_VALUE, max = Number.MIN_VALUE;
    let minSum=0, maxSum=0;
    for(let num of arr){
        totalSum+=num;
        if(num<min) min = num;
        if(num>max) max = num;
    }
    minSum = totalSum-max;
    maxSum = totalSum-min;
    console.log(minSum, maxSum);
}

function main() {

    const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    miniMaxSum(arr);
}
